src/services/ContributingDeveloperAuditService/scm/GitHub/api/GitHubApiClient.ts

```diff
@@ -0,0 +1,218 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { soosLogger } from "../../../../../logging/SOOSLogger";
+import { sleep } from "../../../../../utilities";
+
+interface IHttpRequestParameters {
+  baseUri: string;
+  githubPAT: string;
+}
+
+interface IHttpClientParameters extends IHttpRequestParameters {
+  apiClientName: string;
+}
+
+export interface GitHubOrganization {
+  login: string;
+}
+
+export interface GitHubRepository {
+  id: number;
+  name: string;
+  full_name: string;
+  private: boolean;
+  owner: GitHubOrganization;
+}
+
+export interface ContributingDeveloper {
+  username: string;
+  repositories: ContributingDeveloperRepositories[];
+}
+
+interface ContributingDeveloperRepositories {
+  id: number;
+  name: string;
+  lastCommit: string;
+  isPrivate: boolean;
+}
+
+export interface Commits {
+  commit: {
+    author: Author;
+  };
+}
+
+export interface Author {
+  name: string;
+  email: string;
+  date: string;
+}
+
+export const threeMonthsDate = (() => {
```

Comment:
Might be worth to create a constants file under GitHub (like utilities), with this one and the api base url.

Reply:
Yup, if you need to go for it - that's the pattern for a service folder. Each can have it's own enum.ts/contstants.ts